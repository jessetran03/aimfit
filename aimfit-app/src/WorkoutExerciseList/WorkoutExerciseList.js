import React, { Component } from 'react'
import './WorkoutExerciseList.css'
import STORE from '../STORE'
import WorkoutExercise from '../WorkoutExercise/WorkoutExercise'

export default class WorkoutExerciseList extends Component {
  state = {
    exercises: STORE.exercises,
    workouts: STORE.workouts,
  }
  static defaultProps = {
    match: {
      params: {}
    }
  }

  handleClickAdd = () => {

  }

  handleClickBack = () => {
    this.props.history.goBack()
  };

  render() {

    const { workoutId } = this.props.match.params
    const exercises = this.state.exercises
    const exercisesForWorkouts = exercises.filter(exercise =>
      parseInt(exercise.workoutId) === parseInt(workoutId)
    )
    const workouts = this.state.workouts
    const workoutName = workouts.find(workout =>
      workout.id === parseInt(workoutId)
    ).title
    return (
      <>
        <section className='exercises'>
          <button
            onClick={this.handleClickBack}
          >
            Back
          </button>
          <h2>{workoutName}</h2>
          <ul>
            {exercisesForWorkouts.map(exercise => (
              <li
                key={exercise.id}
              >
                <WorkoutExercise
                  id={exercise.id}
                  name={exercise.name}
                />
              </li>
            ))}
          </ul>
          <button
            onClick={this.handleClickAdd}
          >
            +Add Exercise
          </button>
        </section>
      </>
    )
  }
}