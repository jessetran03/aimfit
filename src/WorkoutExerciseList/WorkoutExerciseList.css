.exercises li {
  background-image: radial-gradient(#FFFFFF, #aab5be);
  padding: 15px 0;
  margin: 10px;
  border-radius: 30px;
  border: 3px solid #000000;
  max-width: 400px;
  margin-left: auto;
  margin-right: auto;
  list-style-type: none;
}
