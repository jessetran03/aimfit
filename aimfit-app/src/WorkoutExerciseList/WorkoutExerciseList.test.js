import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import WorkoutExerciseList from './WorkoutExerciseList'
import renderer from 'react-test-renderer'

it('renders WorkoutExerciseList without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <WorkoutExerciseList />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render WorkoutExerciseList without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <WorkoutExerciseList />
    </BrowserRouter>
  )
    .toJSON();
    expect(wrapper).toMatchSnapshot()
  })