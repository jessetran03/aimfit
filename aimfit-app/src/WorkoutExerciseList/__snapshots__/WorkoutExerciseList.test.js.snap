// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render WorkoutExerciseList without data 1`] = `
<section
  className="exercises"
>
  <button
    onClick={[Function]}
  >
    Back
  </button>
  <h2 />
  <ul />
  <a
    href="/[object Object]/exercises"
    onClick={[Function]}
  >
    <button>
      +Add Exercise
    </button>
  </a>
</section>
`;
